modules/core/04-channel/v2/types/merkle.go

```diff
@@ -1,31 +1,20 @@
 package types
 
 import (
-	commitmenttypes "github.com/cosmos/ibc-go/v9/modules/core/23-commitment/types"
 	commitmenttypesv2 "github.com/cosmos/ibc-go/v9/modules/core/23-commitment/types/v2"
 )
 
 // BuildMerklePath takes the merkle path prefix and an ICS24 path
 // and builds a new path by appending the ICS24 path to the last element of the merkle path prefix.
-func BuildMerklePath(prefix commitmenttypesv2.MerklePath, path []byte) commitmenttypesv2.MerklePath {
-	if prefix.Empty() {
-		return commitmenttypes.NewMerklePath(path)
+func BuildMerklePath(prefix [][]byte, path []byte) commitmenttypesv2.MerklePath {
+	prefixLength := len(prefix)
+	if prefixLength == 0 {
+		panic("cannot build merkle path with empty prefix")
```

Comment:
Don't think this should necessarily error. If there's no prefix, the path should just be returned as it was originally. See removed diff

This would be equivalent to a chain storing the ICS24 paths directly under root. Which is potentially possible for a solomachine chain or something equivalent

Reply:
What is the reasoning for changing the behavior here?

Reply:
Ahh i see it was decided in the original issue! Ok, i mean i suppose we could require all implementations to store under a prefix

Reply:
I don't remember the entire context for why we said this, but it was something we mentioned during the alpha audit itself, I believe.

Reply:
I could take this back. I don't feel strongly about it. What should we do?

Reply:
@AdityaSripal can have the final say on it :)

Reply:
I think its more likely that someone accidentally leaves prefix empty than having a counterparty that is genuinely storing everything under root.

So lets have the error for now, and if we ever have a user run into this issue and report it we can change the behavior of this function quite easily